### ISSUE TITLE
Use HttpClient instead of WebClient

### ISSUE THREAD

**Issue by sabihoshi** [Sylveon76/NHentaiAPI#4]
WebClient is a bit outdated, and doesn't allow parallel calls unless you create multiple instances of it. This makes it unconventional to do parallel http calls. Doing it directly would return an `WebClient does not support concurrent I/O operations` exception.

**Comment by andy840119** [comment]
Is there any recommend package?

**Comment by sabihoshi** [comment]
These two links might be able to help:
https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=netframework-4.8
https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/

**Comment by sabihoshi** [comment]
I was able to do it by overriding the methods simply like so:
```cs
public class NhentaiService : NHentaiClient
{
    private readonly HttpClient _client;

    public NhentaiService(HttpClient client)
    {
        _client = client;
    }

    public override Task<Book> GetBookAsync(int bookId)
    {
        return GetData<Book>(GetBookDetailsUrl(bookId));
    }

    protected override async Task<TOutput> GetData<TOutput>(string rootUrl)
    {
        TOutput output;
        try { output = JsonConvert.DeserializeObject<TOutput>(await _client.GetStringAsync(rootUrl)); }
        catch (Exception ex) { throw ex; }

        return output;
    }
}
```
However the only problem with this is that it does not include an internal ratelimiter so I get 503 errors if I request too much.

**Comment by igorquintaes** [comment]
> I was able to do it by overriding the methods simply like so:
> 
> ```cs
> public class NhentaiService : NHentaiClient
> {
>     private readonly HttpClient _client;
> 
>     public NhentaiService(HttpClient client)
>     {
>         _client = client;
>     }
> 
>     public override Task<Book> GetBookAsync(int bookId)
>     {
>         return GetData<Book>(GetBookDetailsUrl(bookId));
>     }
> 
>     protected override async Task<TOutput> GetData<TOutput>(string rootUrl)
>     {
>         TOutput output;
>         try { output = JsonConvert.DeserializeObject<TOutput>(await _client.GetStringAsync(rootUrl)); }
>         catch (Exception ex) { throw ex; }
> 
>         return output;
>     }
> }
> ```
> 
> However the only problem with this is that it does not include an internal ratelimiter so I get 503 errors if I request too much.

In the case of nonconsecutive 503 errors, you can use recursion approach to avoid some unsuccessful status codes:

```cs
protected override async Task<TOutput> GetData<TOutput>(string rootUrl)
{
    var httpResult = await _client.GetAsync(rootUrl);

    return httpResult.IsSuccessStatusCode
         ? JsonConvert.DeserializeObject<TOutput>(await httpResult.Content.ReadAsStringAsync())
         : await GetData<TOutput>(rootUrl);
}
```

**Comment by andy840119** [comment]
Sorry i'm busy recently :0
I'll spend some time searching about difference between HttpClient and WebClient

**Comment by andy840119** [comment]
@hizamakura Thanks to your feedback.
I fixed it on #5 
Feel free to make a issue or PR if you need : )
